pyomo/repn/plugins/gams_writer.py

```diff
@@ -201,46 +312,62 @@ def _print_model( self, model, output_file, solver_capability, var_list,
             var = var_list[i]
             i += 1
             v = symbolMap.getObject(var)
-            try:
+            if not v.is_expression():
                 if v.is_binary():
                     binary.append(var)
                 elif v.is_integer():
                     if v.bounds == (0,1):
                         binary.append(var)
+                    elif v.lb is not None and v.lb >= 0:
+                        posInts.append(var)
                     else:
-                        ints.append(var)
+                        otherInts.append(var)
                 elif v.lb == 0:
                     positive.append(var)
                 else:
                     reals.append(var)
-            except AttributeError:
-                # This is an Expression object
-                body = StringIO()
-                v.to_string(body)
+            else:
+                # Same domain violation check as above
+                try:
+                    value(v.expr)
+                except:
+                    raise RuntimeError(
+                        "While evaluating: %s\n\tProblemWriter_gams"
+                        " encountered an error. Ensure initial"
+                        " variable values\n\tdo not violate"
+                        " any domains (are you using log/log10?)"
```

Comment:
This is the domain check for Expressions, which also exhibit the initial value domain restriction in GAMS, so it was necessary to check them too. Pyomo does make sure variables are set within their bounds, but expressions may have a log function that it's constituent variables violate even if they're inside their bounds. As for the log/log10 comment, those are still the only functions I know of that have this restriction.